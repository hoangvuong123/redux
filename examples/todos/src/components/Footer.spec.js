import React from 'react';
import { shallow } from 'enzyme';
import Footer from './Footer';

const setup = () => {
  const wrapper = shallow(<Footer/>);
  return { wrapper };
};

describe('Footer component', () => {
  it('Should render Footer component', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  })
})
