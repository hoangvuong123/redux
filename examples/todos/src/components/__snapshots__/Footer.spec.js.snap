// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Should render Footer component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span>
          Show: 
</span>,
        <Connect(Link)
          filter="SHOW_ALL"
>
          All
</Connect(Link)>,
        <Connect(Link)
          filter="SHOW_ACTIVE"
>
          Active
</Connect(Link)>,
        <Connect(Link)
          filter="SHOW_COMPLETED"
>
          Completed
</Connect(Link)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Show: ",
        },
        "ref": null,
        "rendered": "Show: ",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "All",
          "filter": "SHOW_ALL",
        },
        "ref": null,
        "rendered": "All",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Active",
          "filter": "SHOW_ACTIVE",
        },
        "ref": null,
        "rendered": "Active",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Completed",
          "filter": "SHOW_COMPLETED",
        },
        "ref": null,
        "rendered": "Completed",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            Show: 
</span>,
          <Connect(Link)
            filter="SHOW_ALL"
>
            All
</Connect(Link)>,
          <Connect(Link)
            filter="SHOW_ACTIVE"
>
            Active
</Connect(Link)>,
          <Connect(Link)
            filter="SHOW_COMPLETED"
>
            Completed
</Connect(Link)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Show: ",
          },
          "ref": null,
          "rendered": "Show: ",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "All",
            "filter": "SHOW_ALL",
          },
          "ref": null,
          "rendered": "All",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Active",
            "filter": "SHOW_ACTIVE",
          },
          "ref": null,
          "rendered": "Active",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Completed",
            "filter": "SHOW_COMPLETED",
          },
          "ref": null,
          "rendered": "Completed",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
